src/hooks/project-hooks.tsx

```diff
@@ -99,6 +111,18 @@ export const ProjectProvider = ({
   const getCurrentProject = useStore((s) => s.getCurrentProject);
   const setPostImportDialogState = useStore((s) => s.setPostImportDialogState);
   const navigate = useNavigate();
+
+  const projectName = useStore((s) => s.project.header?.name);
+  const setProjectName = useStore((s) => s.setProjectName);
+  const defaultProjectName = useDefaultProjectName();
+
+  // If project name is null, set project name as translated "untitled" name.
+  useEffect(() => {
+    if (projectName === null) {
+      setProjectName(defaultProjectName);
+    }
+  }, [defaultProjectName, intl, projectName, setProjectName]);
```

Comment:
Not sure if this is the best way of doing it. The store can't access the translated untitled string

Reply:
Hmm this has ended up a bit unpleasant. Is it feasible to pass it in? I guess initialisation might be awkward - store init would need to be triggered from in a React context and we don't have a notion of waiting for init right now?

Reply:
Perhaps OK to use English "Untitled" or another placeholder and swap at the point of asking for a name for the project?

Reply:
I think delaying the switch between using the translated untitled version makes sense 👍 I have made the change in this commit [f90f00d](https://github.com/microbit-foundation/ml-trainer/pull/462/commits/f90f00dd80af591e9e8f144be2aabcc43c91ae11) and have had follow-up commits to clean up/fix/revert other bits

Reply:
I'd still like to play with this some more and see if there's a simpler option - but if I don't get to it today we'll go ahead with this.

Reply:
Had a play, can't see a reasonable-effort approach to improving this. Sorry for the pointless delay.